packages/next/src/server/app-render/app-render.tsx

```diff
@@ -2834,6 +2816,12 @@ async function prerenderToStream(
         initialServerRenderController.abort()
         initialServerPrerenderController.abort()
 
+        // We don't need to continue the prerender process if we already
+        // detected invalid dynamic usage in the initial prerender phase.
+        if (workStore.invalidDynamicUsageError) {
+          throw workStore.invalidDynamicUsageError
+        }
+
```

Comment:
does this make sense even when doNotThrowOnEmptyStaticShell is set? We could interrupt the prerender but we probably don't want to fail the build here

Reply:
Yeah, this will change in #78882, as I mentioned in the PR description:
>In a future PR, we will adapt the behaviour for prerendering of fallback shells that are allowed to be empty, in which case the timeout must not fail the build.